iib/workers/tasks/build.py

```diff
@@ -337,6 +341,35 @@ def _get_resolved_image(pull_spec):
     return pull_spec_resolved
 
 
+def _serve_index_db(from_index, base_dir):
+    """
+    image
+    """
+    cidfile = os.path.join(base_dir, "cidfile")
+    run_cmd(["podman", "create", "--cidfile", cidfile, from_index], exc_msg='Failed to create index image container')
+    with open(cidfile, 'r') as f:
+        cid = f.read()
+    os.remove(cidfile)
+
+    data = json.loads(run_cmd(["podman", "image", "inspect", from_index], exc_msg='Failed to fetch image metadata'))
+    db_path = data[0]["Config"]["Labels"]["operators.operatorframework.io.index.database.v1"]
+
+    run_cmd(["podman", "cp", "{0}:{1}".format(cid, db_path), base_dir], exc_msg='Failed to copy db file from container')
+    run_cmd(["podman", "rm", cid], exc_msg='Failed to delete container')
+    local_path = os.path.join(base_dir, db_path.split("/")[-1])
+    rpc_proc = subprocess.Popen(["opm","registry", "serve", "-p", '50051', "-d", local_path])
```

Comment:
How do you know port 50051 is open? You need to pick an open port since multiple workers on the same VM could be using the port or even some other process unrelated to IIB.

Reply:
I have added port availability check